src/app/pages/questions/components/question/text-input/text-input.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+
+import { LocalizationService } from '../../../../../core/services/misc/localization.service'
+
+@Component({
+  selector: 'text-input',
+  templateUrl: 'text-input.component.html'
+})
+export class TextInputComponent implements OnInit {
+  @Output()
+  valueChange: EventEmitter<string> = new EventEmitter<string>()
+  @Input()
+  type: string
+
+  showDatePicker: boolean
+  showTimePicker: boolean
+  showDurationPicker: boolean
+  showTextInput = true
+  showSeconds: boolean
+
+  datePickerValues: { [key: string]: string[] }
+  defaultDatePickerValue: { [key: string]: string }
+  timePickerValues: { [key: string]: string[] }
+  defaultTimePickerValue: { [key: string]: string }
+  durationPickerValues: { [key: string]: string[] }
+  defaultDurationPickerValue: { [key: string]: string }
+  labels = {
+    day: 'Day',
+    month: 'Month',
+    year: 'Year',
+    hour: 'Hour',
+    minute: 'Minute',
+    second: 'Second',
+    ampm: 'AM/PM'
+  }
+
+  value = {}
+
+  constructor(private localization: LocalizationService) {}
+
+  ngOnInit() {
+    if (this.type.length) {
+      this.showTextInput = false
+      this.showDatePicker = this.type.includes('date')
+      this.showTimePicker = this.type.includes('time')
+      this.showDurationPicker = this.type.includes('duration')
+    }
+    this.showSeconds = this.type.includes('second')
+    this.initValues()
+  }
+
+  initValues() {
+    if (this.showDatePicker) this.initDates()
+    if (this.showTimePicker) this.initTime()
+    if (this.showDurationPicker) this.initDuration()
+  }
+
+  initDates() {
+    const moment = this.localization.moment()
+    const locale = moment.localeData()
+    const month = locale.monthsShort()
+    const day = this.addLeadingZero(Array.from(Array(32).keys()).slice(1, 32))
+    const year = Array.from(Array(31).keys()).map(d => String(d + 2000))
+    this.datePickerValues = { day, month, year }
+    this.defaultDatePickerValue = {
+      day: moment.format('DD'),
+      month: moment.format('MMM'),
+      year: moment.format('YYYY')
+    }
+  }
+
+  initTime() {
+    const moment = this.localization.moment()
```

Comment:
There is still some problem here, the default values don't get updated from one questionnaire to the next, unless the app is restarted. I think it's because the localization service returns a locally saved moment instead of a new one with the `moment()` call, and the saved one only gets updated in the constructor or on language change.

But I'm not sure if this is wanted behaviour or should be fixed in the localization service...

Reply:
Note that you can pass `localization.moment(Date.now())` to get a localised moment for the current time.

Reply:
Right, thanks!